File: backend/src/main/java/t/lab/guide/dto/point/category/CategoryListResponse.java
package t.lab.guide.dto.point.category;

import io.swagger.v3.oas.annotations.media.ArraySchema;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;

import java.util.List;

@Schema(description = "Ответ с списком категорий")
@Builder
public record CategoryListResponse (
    @ArraySchema(
            arraySchema = @Schema(description = "Список категорий"),
            schema = @Schema(implementation = CategoryItem.class)
    )
    List<CategoryItem> categories
){}

File: backend/src/main/kotlin/t/lab/guide/dto/point/category/CategoryListResponse.kt
package t.lab.guide.dto.point.category

import io.swagger.v3.oas.annotations.media.ArraySchema
import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Ответ с списком категорий")
data class CategoryListResponse(
    @ArraySchema(
        arraySchema = Schema(description = "Список категорий"),
        schema = Schema(implementation = CategoryItem::class),
    )
    val categories: List<CategoryItem>,
)
